perf(allocator/vec2): reorder `RawVec` fields

Dunqing opened the pull request:
According to `len` is more commonly used than `cap` and `a` is used less than any other field,  reorder `RawVec` fields to get the best performance. 

<img width="721" alt="image" src="https://github.com/user-attachments/assets/c044e134-2d28-45a4-8d8d-91ec5166612f" />


I've tried putting `cap` before `len`, the performance is not as good as the current order, see https://github.com/oxc-project/oxc/pull/11050#issuecomment-2883194443

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11050?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#11050** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11050?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11050?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10884** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10884?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#10883** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10883?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11050).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
```rs
#[repr(C)]
pub struct RawVec<'a, T> {
    ptr: NonNull<T>,
    cap: u32,
    pub(super) len: u32,
    a: &'a Bump,
}
```
<img width="616" alt="image" src="https://github.com/user-attachments/assets/eda2c47b-8ced-4507-a1e7-e927c8e7d39a" />

overlookmotel commented:
### Merge activity

* **May 15, 6:31 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
perf(allocator/vec2): reorder `RawVec` fields (#11050)

According to `len` is more commonly used than `cap` and `a` is used less than any other field,  reorder `RawVec` fields to get the best performance.

<img width="721" alt="image" src="https://github.com/user-attachments/assets/c044e134-2d28-45a4-8d8d-91ec5166612f" />

I've tried putting `cap` before `len`, the performance is not as good as the current order, see https://github.com/oxc-project/oxc/pull/11050#issuecomment-2883194443
